src/js/components/products/detail.tsx

```diff
@@ -34,6 +56,17 @@ const ProductDetail = ({ product, productSlug, doFetchObject }: Props) => {
     }
   }, [product, productSlug, doFetchObject]);
 
+  useEffect(() => {
+    if (product && (!projects || !projects.fetched)) {
+      // Fetch projects from API
+      doFetchObjects({
+        objectType: OBJECT_TYPES.PROJECT,
+        filters: { product: product.id },
+        reset: true,
+      });
+    }
+  }, [product, projects, doFetchObjects]);
```

Comment:
I think I don't adequately understand `useEffect`; what are the values in the trailing array, and how does this set of them make it behave as we expect?

Reply:
@wlonk It's optional and it basically tells React: I only want to run this function if the values in this array have changed.

Reply:
Technically optional, but strongly advised: https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects